pysindy/optimizers/stlsq.py

```diff
@@ -145,7 +145,7 @@ def _sparse_coefficients(self, dim, ind_nonzero, coef, threshold):
             c[~big_ind] = 0
         return c, big_ind
 
-    def _regress(self, x, y, dim, sparse_sub):
+    def _regress(self, x, y, dim, existing_vals):
```

Comment:
Another NIT... here, the variable makes sense with the name `sparse_sub`, as this is a function that knows it's role in representing a subset of indices to sparsify.

`_remove_and_decrement`, on the other hand, deals with a generic 1-D array of values.  So within that scope, it calls the argument `existing_vals`

The naming of variables reflects what role the variable plays within the scope, which also helps organize code and tests.